src/utility/HCISharedMemTransport.cpp

```diff
@@ -0,0 +1,837 @@
+{
+    sys_resp = false;
+    return;
+}
+
+void shci_cmd_resp_wait(uint32_t timeout)
+{
+    /* TO DO: manage timeouts if we can return an error */
+//    for (unsigned long start = millis(); (millis() - start) < 10000;) {
+    while(1) {
+        /* Wait for 10sec max - if not return an error */
+        if (sys_resp) { break; }
+    }
+
+    if (!sys_resp) {
+        /*  no event received, timeout occurs */
+        return;
+    }
+}
+
+static void acl_data_ack(void)
+{
+    /**
+     * The current implementation assumes the tackGUI will not send a new HCI ACL DATA packet before this ack is received
+     * ( which means the CPU2 has handled the previous packet )
+     * In order to implement a secure mechanism, it is required either
+     * - a flow control with the stack
+     * - a local pool of buffer to store packets received from the stack
+     */
+    acl_data = false;
+    return;
+}
+
+static bool acl_data_wait(void)
+{
+    /* Wait 10 sec for previous ACL command to be ack'ed by Low Layers
+     * before sending the next one */
+//    for (unsigned long start = millis(); (millis() - start) < 10000;) {
+    while(1) {
+        /* Wait for 10sec max - if not return an error */
+        if (acl_data) { break; }
+    }
+        if (!acl_data) {
+        /*  no event received, timeout occurs */
+         return false;
+    } else {
+        /*  release immmediately, now that M0 runs */
+        acl_data = false;
+        return true;
+    }
+}
+
+static void syscmd_status_not(SHCI_TL_CmdStatus_t status)
+{
+    return;
+}
+
+void shci_notify_asynch_evt(void *pdata)
+{
+    /* Need to parse data in future version */
+    shci_user_evt_proc();
+    return;
+}
+
+void shci_register_io_bus(tSHciIO *fops)
+{
+    /* Register IO bus services */
+    fops->Init    = TL_SYS_Init;
+    fops->Send    = TL_SYS_SendCmd;
+}
+
+HCISharedMemTransportClass::~HCISharedMemTransportClass()
+{
+}
+
+int HCISharedMemTransportClass::begin()
+{
+  /* prepare data */
+  _read_index = 0;
+  _write_index = 0;
+  _write_index_initial = 0;
+
+  memset(_rxbuff, 0, sizeof(_rxbuff));
+
+  start_ble_rf();
+
+        /*  Check whether M0 sub-system was started already by
+         *  checking if the system event was already received
+         *  before. If it was not, then go thru all init. */
+        if (!sys_event) {
+            init_debug();
+            /* Take BLE out of reset */
+            stm32wb_reset();
+            /* "C2 unlocking" */
+            transport_init();
+            /*  At this stage, we got the ready event,
+             *  passed thru TL_SYS_EvtReceived */
+
+            WirelessFwInfo_t wireless_info_instance;
+            WirelessFwInfo_t *p_wireless_info = &wireless_info_instance;
+            SHCI_GetWirelessFwInfo(p_wireless_info);
+            /* 
+             * tr_info("WIRELESS COPROCESSOR FW VERSION ID = %d.%d.%d", p_wireless_info->VersionMajor, p_wireless_info->VersionMinor, p_wireless_info->VersionSub);
```

Comment:
Why not tracing he info ? or is it done at another place now ?

Reply:
p_wireless_info 
= {VersionMajor = 0x1
, VersionMinor = 0x8
, VersionSub = 0x0
, VersionBranch = 0x0
, VersionReleaseType = 0x4
, MemorySizeSram2B = 0x11
, MemorySizeSram2A = 0x16
, MemorySizeSram1 = 0x0
, MemorySizeFlash = 0x29
, StackType = 0x1
, FusVersionMajor = 0x1
, FusVersionMinor = 0x1
, FusVersionSub = 0x0
, FusMemorySizeSram2B = 0x10

Reply:
trace is available at upper level

Reply:
ok then I f=suggest to remove unused code

Reply:
printf the value with debug option